### ISSUE TITLE
initialize is async, it should return a Promise

### ISSUE THREAD

**Issue by JSteunou** [OfficeDev/microsoft-teams-library-js#172]
Have the case sometimes with authentification process. Calling initialize, then authenticate, but `hostClientType` is not set yet because called to soon. Adding a setTimeout do the trick but that's not pretty nor reliable.

**Comment by ydogandjiev** [comment]
Thanks @JSteunou for reporting this issue! We’ve been aware of it for sometime and will attempt to address it in the future; however, the recommended approach for tabs is to always call microsoftTeams.initialize() in their page’s onload method or shortly thereafter. On the other hand, the microsoftTeams.authentication.authenticate() call should generally be made only in response to the user clicking on a “Sign in” button (in order to avoid triggering browsers’ popup blocker). This typically provides more than enough time for the hostClientType to get determined.

**Comment by JSteunou** [comment]
Yes, but I have this case #171 where I'm using `authenticate` to open a popup (because `window.open`) does not work in Electron, and I do not want to spread the MS logic everywhere in my app, so I'm calling `initialize` in the same logic code that is using `authenticate`, like just a line before. 

In that specific case I would need a Promise back :)

**Comment by Tyrcheg** [comment]
@JSteunou Hi, why not to wrap initialize inside of new Promise()?

**Comment by JSteunou** [comment]
I could, but when I should call `resolve` ? Using a `setTimeout` ? This is not realistic.

**Comment by ydogandjiev** [comment]
I've sent out a PR to address this issue here - #249. @JSteunou, please take a look and let me know if you think that it will suit your needs.

**Comment by ydogandjiev** [comment]
Apps can now wait for initialization to complete as follows:
```
microsoftTeams.initialize(() => {
  // Initialization complete
});
```

Or if a promise-based solution is required like so:
```
return new Promise(resolve => {
  microsoftTeams.initialize(() => {
    resolve();
  });
});
```

This change will be released in v1.5 of our SDK.
